.changeset/empty-fans-reflect.md

```diff
@@ -0,0 +1,65 @@
+---
+'astro': minor
+---
+
+Adds a new `experimental.typescript` configuration option
```

Comment:
Just noting before I get into a full review that this name is pretty generic, and can make it look like *TypeScript* is experimental in Astro. :joy:

Usually these names are specific to what they do. If this is to provide more fine-grained TS config options, maybe even `experimental.tsconfig` which makes it super clear that this is a new way to "handle" these options!

Reply:
Good point! This is inspired by nuxt which have a `typescript` option but it makes more stuff so yeah I think `experimental.tsconfig` makes sense. I'll update

Reply:
Done!